src/Substrait.Core/ProtoConverter.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Substrait.Expression;
+using Substrait.Protobuf;
+using Substrait.Relation;
+using System.Collections.Immutable;
+using ProtoRel = Substrait.Protobuf.Rel;
+using ProtoRex = Substrait.Protobuf.Expression;
+using Rel = Substrait.Relation.Rel;
+
+namespace Substrait.Core
+{
+  /// <summary>
+  /// Utility to convert Substrait protobuf messages to/from charp domain-objects.
+  /// </summary>
```

Comment:
Can you please more details on the motivation to add a `Substrait.Relation.Rel` mappings for the protobuf objects? I see some of it in the [previous PR](https://github.com/substrait-io/substrait-csharp/pull/2#issuecomment-1339775232), trying to learn more.

Reply:
One very useful feature is that `Rel` has a property `Inputs`:

```
    public IList<Rel> Inputs { get; init; } = new List<Rel>();
```

This can be used, for example, to do a depth-first traversal of the relations tree.  However, in the protobuf objects, there is no such property.  For example, a join relation has `left` and `right` properties and a filter relation has an `input` property.

Reply:
Ok.
Is the intention for C# users to use these objects to interface with substrait instead of the protofbuf objects when it is fully ready?